doc/manual/castxml.1.rst

```diff
@@ -99,3 +99,52 @@ case, CastXML always adds the following predefined macros:
 
 Source files may use these to identify the tool that is actually doing the
 preprocessing even when ``--castxml-cc-<id>`` changes the predefined macros.
+
+FAQ
+===
+
+Why are C++ function bodies not dumped in XML?
+----------------------------------------------
+
+The original sponsors of the project had no need for function bodies.
+So far the authors have not had time or funding to implement the support.
+
+Is there a DTD specifying the XML format dumped?
+------------------------------------------------
+
+No.
+
+Why are most classes, structs, and enums marked as "artificial"?
+----------------------------------------------------------------
+
+The artificial attribute marks nodes that are artificially generated by the compiler
+(such as implicit copy constructors). The C++ code
+
+ | struct foo {};
+
+is almost the same as the C code
+
+ | struct foo {};
+ | typedef struct foo foo;
+
+because the C++ compiler adds the typedef artificially.
+The name foo for the struct shown above is reported as the name in the Struct
+element in the XML output, but this name was artificially generated so that
+programmers would not have to write struct foo.
```

Comment:
I just double checked. Constructors and destructors are marked as artificial for
```
struct foo {};
```
, as you said, and which makes sense.

Did you mean that in GCCXML you had something like this:
```
  <Struct id="_6" name="foo" .... artificial="1"/>
```
? This his is now gone. I am OK on removing that FAQ entry.